src/lightning-address.ts

```diff
@@ -82,26 +82,31 @@ export default class LightningAddress {
     if (!this.domain || !this.username) {
       return;
     }
-    const lnurlResult = await fetch(this.lnurlpUrl());
-    const keysendResult = await fetch(this.keysendUrl());
-    const nostrResult = await fetch(this.nostrUrl());
 
-    let lnurlData: LnUrlRawData | undefined;
-    if (lnurlResult.ok) {
-      lnurlData = await lnurlResult.json();
-    }
-    let keysendData: KeySendRawData | undefined;
-    if (keysendResult.ok) {
-      keysendData = await keysendResult.json();
-    }
-    let nostrData: NostrResponse | undefined;
-    if (nostrResult.ok) {
-      nostrData = await nostrResult.json();
-    }
+    const [lnurlData, keysendData, nostrData] = await Promise.all([
+      this.requestLnurlData(),
+      this.requestKeysendData(),
+      this.requestNostrData(),
+    ]);
 
     await this.parseResponse(lnurlData, keysendData, nostrData);
   }
 
+  async fetchLnurlData() {
```

Comment:
Minor: `fetch` and `request` names are similar to me and it creates confusion because one relies on the other. Could we just move the code from the request functions into these ones instead?

Reply:
fetch is public, request is private and called from two places.

Reply:
fetchWithoutProxy calls parseResponse with all three params... maybe `parseResponse` should also be refactored in three...

Reply:
refactored parseResponse and removed request methods